crates/re_renderer/src/error_handling/wgpu_error_scope.rs

```diff
@@ -0,0 +1,39 @@
+use std::sync::Arc;
+
+/// Wgpu device error scope for all filters that auto closes when exiting the scope unless it was already closed.
+///
+/// The expectation is that the scope is manually closed, but this construct is useful to not accidentally
+/// leave the scope open when returning early from a function.
+/// Opens scopes for all error types.
+pub struct WgpuErrorScope {
+    open: bool,
+    device: Arc<wgpu::Device>,
```

Comment:
It may be worth forbidding `Send` on this scope, which can accidentally happen with `async` code (an `.await` can result in continuation on a new thread!)

```suggestion
    device: Arc<wgpu::Device>,
    _not_send: std::marker::PhantomData<*const ()>,
```

Reply:
My first reaction was that this is a great idea. Buuuut that would mean that `RendererContext` which contains one of these scopes would no longer be `Send`. So we can't have this!